lib/tests/modules/freeform-submodules.nix

```diff
@@ -3,6 +3,9 @@
   options.fooDeclarations = lib.mkOption {
     default = (options.free.type.getSubOptions [])._freeformOptions.foo.declarations;
   };
+  options.fooDeclarationsLoc = lib.mkOption {
+    default = (options.free.type.getSubOptions [])._freeformOptions.foo.declarationsWithLocations;
+  };
```

Comment:
This doesn't seem like the right place to test this since it's not really related to freeform modules. Instead you can add more modules in the `lib/tests/modules` directory and call those from the test script.

Reply:
moved.